coderd/workspaces.go

```diff
@@ -681,15 +682,32 @@ func createWorkspace(
 				Ttl:               dbTTL,
 				// The workspaces page will sort by last used at, and it's useful to
 				// have the newly created workspace at the top of the list!
-				LastUsedAt:       dbtime.Now(),
+				LastUsedAt:       now,
 				AutomaticUpdates: dbAU,
 			})
 			if err != nil {
 				return xerrors.Errorf("insert workspace: %w", err)
 			}
 			workspaceID = minimumWorkspace.ID
 		} else {
-			// Prebuild found!
+			// Prebuild found! Update lifecycle related parameters
+			err = db.UpdateWorkspaceAutostart(ctx, database.UpdateWorkspaceAutostartParams{
+				ID:                claimedWorkspace.ID,
+				AutostartSchedule: dbAutostartSchedule,
+				NextStartAt:       nextStartAt,
+			})
+			if err != nil {
+				return xerrors.Errorf("update claimed workspace Autostart: %w", err)
+			}
+
+			err = db.UpdateWorkspaceTTL(ctx, database.UpdateWorkspaceTTLParams{
+				ID:  claimedWorkspace.ID,
+				Ttl: dbTTL,
+			})
+			if err != nil {
+				return xerrors.Errorf("update claimed workspace TTL: %w", err)
+			}
```

Comment:
Perhaps add a comment mentioning what fields get updated by ClaimPrebuiltWorkspace. I can also see an argument for refactoring and passing all of these into `ClaimPrebuiltWorkspace` so it's abundantly obvious.

Reply:
Great idea 🚀 Addressed in https://github.com/coder/coder/pull/19252/commits/8653a33ed22270d3e4020532f665ce03f73ce97f